commons-db/src/main/java/com/palantir/nexus/db/sql/BasicSQLUtils.java

```diff
@@ -53,7 +52,27 @@ public static final int countFields(String fieldList) {
     }
 
     public static String nArguments(int n) {
-        return TextUtils.makeStringList("?", ",", n); //$NON-NLS-1$ //$NON-NLS-2$
+        return makeStringList("?", ",", n); //$NON-NLS-1$ //$NON-NLS-2$
+    }
+
+    /**
+     *  Makes a list of n copies of the given string, separated by the given delimiter.
+     */
+    // TODO this method was copied from TextUtils, which is being moved to an internal product.
+    // If this class follows it there, then the duplication should be removed.
+    private static String makeStringList(String str, String delimiter, int count) {
```

Comment:
Before you change the implementation to use Joiner, you need to write JMH benchmarks showing the difference in execution time and memory allocations & GC overhead. Note that current implementation explicitly sizes the string builder to avoid unnecessary growth, reallocations, and memory copies, which matters when this is part of hot path this becomes painful, see https://github.com/palantir/atlasdb/issues/735 for similar problems